test/debug/control_flow_commands_test.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+require_relative '../support/test_case'
+
+module DEBUGGER__
+  #
+  # Test basic control flow commands.
+  #
+  class BasicControlFlowTest < TestCase
+    def program
+      <<~RUBY
+         1| class Student
+         2|   def initialize(name)
+         3|     @name = name
+         4|   end
+         5|
+         6|   def name
+         7|     @name
+         8|   end
+         9| end
+        10|
+        11| s = Student.new("John")
+        12| s.name
+        13| "foo"
+      RUBY
+    end
+
+    def test_step_goes_to_the_next_statement
+      debug_code(program) do
+        type 'b 11'
+        type 'c'
+        assert_line_num 11
+        type 's'
+        assert_line_num 3
+        type 's'
+        assert_line_num 4
+        type 's'
+        assert_line_num 12
+        type 's'
+        assert_line_num 7
+        type 's'
+        assert_line_num 8
+        type 's'
+        assert_line_num 13
+        type 'quit'
+        type 'y'
+      end
+    end
+
+    def test_next_goes_to_the_next_line
+      debug_code(program) do
+        type 'b 11'
+        type 'c'
+        assert_line_num 11
+        type 'n'
+        assert_line_num 12
+        type 'n'
+        assert_line_num 13
+        type 'quit'
+        type 'y'
+      end
+    end
+
+    def test_continue_goes_to_the_next_breakpoint
+      debug_code(program) do
+        type 'b 11'
+        type 'c'
+        assert_line_num 11
+        type 'b 13'
+        type 'c'
+        assert_line_num 13
+        type 'quit'
+        type 'y'
+      end
+    end
+
+    def test_finish_leaves_the_current_frame
+      debug_code(program) do
+        type 'b 11'
+        type 'c'
+        assert_line_num 11
+        type 's'
+        assert_line_num 3
+        type 'fin'
+        assert_line_num 12
+        type 's'
+        assert_line_num 7
+        type 'fin'
```

Comment:
@ko1 not directly related, but I found it confusing to have `fin` instead of `f` as `finish`'s shortcut. maybe we can change it to make it consistent with other commands?

Reply:
nvm, I found that it's been taken by `frame` 🤦

Reply:
YES.